deps/src/polymake_jlcxx.h

```diff
@@ -0,0 +1,23 @@
+#ifndef POLYMAKE_JLCXX
+#define POLYMAKE_JLCXX
+
+// This must be the very first include
+
+// workaround xcode 11.4 issue until upstream fix arrives in xcode:
+// https://github.com/llvm/llvm-project/commit/2464d8135e
+//
+// first include __config from libc++, then override typeinfo flag
+// to force use of address as hash instead of hashing the string
+
+#if defined(__APPLE__) && defined(FORCE_XCODE_TYPEINFO_MERGED)
+#include <__config>
+#if defined(_LIBCPP_HAS_MERGED_TYPEINFO_NAMES_DEFAULT) && \
+    _LIBCPP_HAS_MERGED_TYPEINFO_NAMES_DEFAULT == 0
```

Comment:
Note that if `_LIBCPP_HAS_MERGED_TYPEINFO_NAMES_DEFAULT` is *not* defined, the above check would result in a compiler warning or even error, as it's expand to `#if false && == 0`.

But of course it should always be defined here, and if it is not, something really funky is going on that we don't know how to deal with, so perhaps this isn't so bad...  *shrug*.

Reply:
The C preprocessor does shortcut and gives no warning (both clang and gcc) when there is a `defined() &&` before the comparison, even with `-Wall -Wundef`. (I built a small test-program)

But I added an `#else #error` just in case something weird is going on.